File: apps/api/app/schemas/person.py
"""
Schemas Pydantic para Personas y Asignaciones de Personal.
"""
import uuid
from datetime import datetime
from pydantic import BaseModel, ConfigDict, EmailStr


class PersonaBase(BaseModel):
    nombre_completo: str
    rut_dni: str | None = None
    email: str | None = None
    telefono: str | None = None
    cargo: str | None = None
    tipo: str = "DOCENTE"
    unidad_id: uuid.UUID | None = None
    user_id: uuid.UUID | None = None
    metadata_extra: str | None = None


class PersonaCreate(PersonaBase):
    pass


class PersonaUpdate(BaseModel):
    nombre_completo: str | None = None
    rut_dni: str | None = None
    email: str | None = None
    telefono: str | None = None
    cargo: str | None = None
    tipo: str | None = None
    unidad_id: uuid.UUID | None = None
    user_id: uuid.UUID | None = None
    metadata_extra: str | None = None


class PersonaRead(PersonaBase):
    model_config = ConfigDict(from_attributes=True)
    id: uuid.UUID
    created_at: datetime
    updated_at: datetime


class PersonaWithUnidadRead(PersonaRead):
    unidad_nombre: str | None = None
    unidad_codigo: str | None = None

File: packages/shared-types/src/entities/person.ts
export interface Persona {
  id: string;
  nombre_completo: string;
  rut_dni?: string | null;
  email?: string | null;
  telefono?: string | null;
  cargo?: string | null;
  tipo: string;
  unidad_id?: string | null;
  user_id?: string | null;
  metadata_extra?: string | null;
  unidad_nombre?: string | null;
  unidad_codigo?: string | null;
  created_at: string;
  updated_at: string;
}
